### PR TITLE
feat: Added Snap to grid functionality. 

### PR DESCRIPTION
**Pull request by severfire**
I added Snap to grid functionality. 
It also adds icon right to Dark mode icon, as it is interface functionality. 
It stores setting in localStorage.

### CODE REVIEW

File: src/frontend/src/stores/gridSnapping.ts [hunk 1]
```diff
@@ -0,0 +1,11 @@
+import { create } from "zustand";
+import { GridStoreType } from "../types/zustand/grid";
+
+export const useGridSnappingStore = create<GridStoreType>((set, get) => ({
+  gridSnapping: JSON.parse(window.localStorage.getItem("isGridSnap")!) ?? false,
```
Comment (anovazzi1) [inline src/frontend/src/stores/gridSnapping.ts:5]: hey, what do you think about this change
```
gridSnapping: window.localStorage.getItem("isGridSnap") ? JSON.parse(window.localStorage.getItem("isGridSnap") : false ,
```

File: src/frontend/src/components/headerComponent/index.tsx [hunk 2]
```diff
@@ -179,6 +185,22 @@ export default function Header(): JSX.Element {
               <IconComponent name="MoonIcon" className="side-bar-button-size" />
             )}
           </button>
+
+          <button
```
Comment (anovazzi1) [inline src/frontend/src/components/headerComponent/index.tsx:189]: Could you please add a tooltip to explain the function of this button?

**Review by anovazzi1 — CHANGES_REQUESTED** [review]

File: src/frontend/src/components/headerComponent/index.tsx (hunk 2 above)
Comment (severfire) [reply src/frontend/src/components/headerComponent/index.tsx:189]: Should I use shadTooltipComponent or something else?

File: src/frontend/src/stores/gridSnapping.ts (hunk 1 above)
Comment (severfire) [reply src/frontend/src/stores/gridSnapping.ts:5]: hi, what do you think about this one? we won't be calling 'getItem' twice
`gridSnapping: JSON.parse(window.localStorage.getItem("isGridSnap") ?? "false"),`

this version handles cases where the item doesn't exist in localStorage

in general what I had i took from 
/src/frontend/src/stores/darkStore.ts
`dark: JSON.parse(window.localStorage.getItem("isDark")!) ?? false,`

Comment (lucaseduoli) [reply src/frontend/src/stores/gridSnapping.ts:5]: its perfect! it can be that

**Pull request closed without merging**